internal/k8s/converter.go

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"fmt"
+	"reflect"
+
+	ingressroutev1 "github.com/projectcontour/contour/apis/contour/v1beta1"
+	projectcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Converter handles conversions between types
+type Converter struct {
+	// scheme holds an initializer for converting Unstructured to a type
+	scheme *runtime.Scheme
+}
+
+// ConvertUnstructured converts an unstructured.Unstructured to typed struct
+// Any other type passed other than unstructured is just returned back to caller
+func (c *Converter) ConvertUnstructured(obj interface{}) (interface{}, error) {
+	if c.scheme == nil {
+		c.scheme = runtime.NewScheme()
+
+		// Setup converter to understand custom CRD types
+		_ = projectcontour.AddKnownTypes(c.scheme)
+		_ = ingressroutev1.AddKnownTypes(c.scheme)
+	}
```

Comment:
Nit: probably a bit cleaner to pull this into a `NewConverter()` API.

Reply:
> suggestion: if you don't care about the result, you doin't have to assign it to anything, ie

Yup I had that originally, but the linter complains about it if you don't handle the error. 

```
internal/k8s/converter.go:39:31: Error return value of `projectcontour.AddKnownTypes` is not checked (errcheck)
                projectcontour.AddKnownTypes(c.scheme)
                                            ^
internal/k8s/converter.go:40:31: Error return value of `ingressroutev1.AddKnownTypes` is not checked (errcheck)
                ingressroutev1.AddKnownTypes(c.scheme)
                                            ^
make: *** [Makefile:102: check-golint] Error 1

Reply:
personally I think this is incorrect, the linter needs to shut up, but do what you think is appropriate.

Reply:
I agree with you, let me see if there's a way around this.

Reply:
FWIW, you can silence the linter in a specific case with a [comment](https://github.com/golangci/golangci-lint#nolint) `// nolint(errcheck)`. Maybe in this case, we should check the error and panic.

Reply:
yeah, you're right, if adding these types to the schema can reasonably fail then we should check the error (hint, check(err) is in scope in this package).

Reply:
I cleaned up the calling code to remove the `error`, it was always returning `nil`, so felt just as weird to check if error.